subsys/net/lib/http/http_server.c

```diff
@@ -904,6 +904,88 @@ int handle_http1_rest_resource(struct http_resource_detail_static *static_detail
 	return 0;
 }
 
+int handle_http1_rest_json_resource(struct http_resource_detail_static *static_detail,
+				    struct http_client_ctx *client)
+{
+	struct http_resource_detail_rest_json *rest_json_detail =
+		CONTAINER_OF(&static_detail->common, struct http_resource_detail_rest_json, common);
+
+	int ret, remaining, offset = rest_json_detail->common.path_len;
+	char *ptr;
+	char tmp[64];
+
+	if (!(static_detail->common.bitmask_of_supported_http_methods & BIT(HTTP_POST))) {
```

Comment:
This only allows the json call if POST is allowed for the resource, but then it allows GET also without checking if GET is actually allowed. The check should be if the actual method is allowed for the resource.